### ISSUE TITLE
automatically determine design_regions from geometry

### ISSUE THREAD

**Issue by stevengj** [NanoComp/meep#2025]
Right now you specify the `design_regions` manually to the `OptimizationProblem`, but it seems like all of this information is already in `sim.geometry`.  Couldn't we just extract it from the geometry by default?

e.g. couldn't it just default to the bounding box of all the geometric objects that use some particular material grid?

(There are still cases where you would want to specify `design_regions` manually.  For example, if you have multiple material grids but only want to optimize over some of them.   But that wouldn't prevent us from having a good default if you don't pass the `design_regions`.)

cc @smartalecH

**Comment by smartalecH** [comment]
> but it seems like all of this information is already in sim.geometry.

Not quite. Note that the `MaterialGrid` can have an arbitrary mesh (i.e. a resolution that doesn't correspond to the meep resolution) in *arbtrary* dimensions (e.g. we can have a 2D block, but the design parameters only span one dimension, or it could have different resolutions in different dimensions.

> But that wouldn't prevent us from having a good default if you don't pass the design_regions

Maybe we can assume that if the user doesn't pass any design regions, then the resolution matches the meep resolution, and material grid dimensionality matches the meep simulation dimensionality.

Still, we would probably want some function that returned the number of design parameters meep computes in this case.

**Comment by stevengj** [comment]
But the material grid itself is stored in the `geometry` (because it is the `material` property of one or more objects).     So you don't need to guess what it is?

**Comment by smartalecH** [comment]
Oh yes, if we keep that the same, then you are correct.
